### ISSUE TITLE
cmd/go: "go test" accepts -test.FLAG where FLAG is any flag accepted by "go build"

### ISSUE THREAD

**Issue by minux** [golang/go#18682]
This behavior always existed from Go 1.0.

For example:
```
$ go test image -test.work
WORK=/tmp/go-build271188011
ok  	image	0.361s
```

Is this expected? I find it quite confusing as there is no -test.work flag documented anywhere and it conflicts with user defined -test.FLAG flags. If this is WAI, we probably should document it.

**Comment by rsc** [comment]
Yes, we should definitely fix this, but I think it's too late for Go 1.9.

**Comment by bcmills** [comment]
I think [CL 211358](https://golang.org/cl/211358) will fix this.
